tests/test_resilientsession.py

```diff
@@ -53,3 +58,36 @@ def test_logging_with_connection_error(self):
     def tearDown(self):
         jira.resilientsession.logging.getLogger().removeHandler(self.loggingHandler)
         del self.loggingHandler
+
+
+@patch("requests.Session.get")
+@patch("time.sleep")
+def test_throttling_error_is_retried(
+    mocked_sleep_method: Mock, mocked_get_method: Mock
+):
+    mocked_throttled_response: Response = Response()
+    mocked_throttled_response.status_code = 429
+    mocked_throttled_response.headers["X-RateLimit-Remaining"] = "1"
+    mocked_get_method.return_value = mocked_throttled_response
+    session: jira.resilientsession.ResilientSession = (
+        jira.resilientsession.ResilientSession()
+    )
+    with pytest.raises(JIRAError):
+        session.get("mocked_url")
+    assert mocked_get_method.call_count == 4
+    assert mocked_sleep_method.call_count == 3
+
+
+@patch("requests.Session.get")
+@patch("time.sleep")
+def test_5xx_error_is_not_retried(mocked_sleep_method: Mock, mocked_get_method: Mock):
```

Comment:
let's `pytest.parametrize()` with the rest of the previous `status_code`s that were tested, just so we can make sure any changes to this behaviour are intentional rather than regressions

Reply:
Done, I added all the status_codes previously present in this class and couple of more that usually shouldn't be retried like 403 and 404

Reply:
Perfect